pkg/sql/sqlbase/rowwriter.go

```diff
@@ -828,29 +828,52 @@ func (ru *RowUpdater) UpdateRow(
 	}
 
 	// Update secondary indexes.
+	// Because columns with inverted indexes can have a variable number of entries for a JSON value, we're going to
+	// make sure that we iterate through all of the new entries and add them to the index, even if they're not replacing
+	// an existing KV entry.
 	for i, newSecondaryIndexEntry := range newSecondaryIndexEntries {
-		secondaryIndexEntry := secondaryIndexEntries[i]
+		// We're making sure that if there are more new index entries for an inverted index than old ones, we will
+		// handle the checks for insertion and deletion correctly.
+		var oldSecondaryIndexEntry IndexEntry
```

Comment:
Are you able to change this logic to

```
if len(oldSecondaryIndexEntries) <= i {
batch.InitPut(newSecondaryIndexEntry.Key, &newSecondaryIndexEntry.Value, true)
}
```

and then not need to make the changes below with the exception of the last bit that deletes extra old entries?